### PR TITLE
option for recons loss + train test eval + remove dtype

### PR DESCRIPTION
**Pull request by Edouard360**
* add reconstruction loss as flag in VAE
* move compute log likelihood as a vae function
* change load_dataset to load_datasets for train and test dataset
* in train: now evaluating marginal log likelihood for train/test every 10 epochs
* remove remaining useless “dtype” casting and call .cuda() only for each batch

Questions:

* maybe there is a cleaner way to handle train/test in only one dataset object
* maybe .cuda() should be called in the __iter__ method of dataset ? (so as not to call it both in the train and test procedure)

### CODE REVIEW

File: scvi/train.py [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
     for epoch in range(n_epochs):
-        for i_batch, (sample_batch, local_l_mean, local_l_var, batch_index) in enumerate(data_loader):
+        for i_batch, (sample_batch, local_l_mean, local_l_var, batch_index) in enumerate(data_loader_train):
             sample_batch = Variable(sample_batch)
             local_l_mean = Variable(local_l_mean)
             local_l_var = Variable(local_l_var)
+            if torch.cuda.is_available():
+                sample_batch.cuda()
+                local_l_mean.cuda()
+                local_l_var.cuda()
```
Comment (jeff-regier) [inline scvi/train.py]: Here I think you need
```
            if torch.cuda.is_available():
                sample_batch = sample_batch.cuda()
                local_l_mean = local_l_mean.cuda()
                local_l_var = local_l_var.cuda()
```
(vae.cuda() mutates vae, but when you call cuda on a Variable, it returns a copy and doesn't mutate the argument)

**Comment by jeff-regier** [comment]
You can partition a dataset using

http://pytorch.org/docs/master/data.html#torch.utils.data.sampler.SubsetRandomSampler 

into training and testing subsets.

e.g.,

```
   tt_split = int(0.9 * len(ds))

    train_loader = DataLoader(ds, batch_size=args.batch_size,
                              num_workers=1, pin_memory=has_cuda,
                              sampler=sampler.SubsetRandomSampler(np.mgrid[tt_split:]))
    test_loader = DataLoader(ds, batch_size=args.batch_size,
                             num_workers=1, pin_memory=has_cuda,
                             sampler=sampler.SubsetRandomSampler(np.mgrid[:tt_split]))
```

**Comment by jeff-regier** [comment]
I suspect we don't want to call `.cuda` in the dataset iter method because that would transfer just one example at a time from the CPU to the GPU. Probably faster to assemble a whole minibatch on the CPU and then transfer it the GPU all at once.
